sdk/aqueduct/artifacts/bool_artifact.py

```diff
@@ -70,23 +68,14 @@ def get(self, parameters: Optional[Dict[str, Any]] = None) -> Union[bool, np.boo
         previewed_artifact = artifact_utils.preview_artifact(
             self._dag, self._artifact_id, parameters
         )
-        if previewed_artifact._get_type() != ArtifactType.BOOL:
-            raise InvalidArtifactTypeException(
-                "Error: the computed result is expected to of type bool, found %s"
-                % previewed_artifact._get_type()
-            )
-
-        assert isinstance(previewed_artifact._get_content(), bool) or isinstance(
-            previewed_artifact._get_content(), np.bool_
-        )
+        content = previewed_artifact._get_content()
+        assert isinstance(content, bool) or isinstance(content, np.bool_)
 
-        if parameters:
-            return previewed_artifact._get_content()
-        else:
-            # We are materializing an artifact generated from lazy execution.
-            assert self._get_content() is None
-            self._set_content(previewed_artifact._get_content())
-            return self._get_content()
+        # If the artifact was previously generated lazily, materialize the contents.
+        if self._get_content() is None:
+            self._set_content(content)
```

Comment:
I think we should set the content only when `parameter == None`. Otherwise when the user previews the artifact again without the parameter, we will just return the version that's evaluated with the custom parameter, which is incorrect.